### ISSUE TITLE
Suggestion: Randomize image order every epoch for gradient accumulation

### ISSUE THREAD

**Issue by Heathen** [aria1th/Hypernetwork-MonkeyPatch-Extension#9]
As it stands, when you use gradient accumulation, every X images are always used together. This only reduces your effective dataset. You do gain the advantage of averaging your dataset during training, but lose potential details.

The best of both worlds would be randomizing which pictures get accumulated together so that the averaging effect isn't always the same every epoch and you get different images working together each time. This would give you the benefit of gradient accumulation plus preserve your dataset size and allow for details bleed through.

As it stands, I have to train with high gradient for a while so the hypernetwork gets a 'smooth' sense of the style I'm training and then set gradient back to 1 with a low learning rate so it can pick up on the details of individual images.

This issue exists in vanilla training as well, but since you're improving training with your extension, I figured I'd drop this suggestion here.

**Comment by aria1th** [comment]
Its already fixed in https://github.com/aria1th/Hypernetwork-MonkeyPatch-Extension/commit/f2c1ea8228ca830c773a086a3f5f08d6c84e31e5, (should)
Can you check it? 
![image](https://user-images.githubusercontent.com/35677394/206644775-7db7f418-71f4-4bc3-a049-d444179a9763.png)

**Comment by Heathen** [comment]
Oh, you are very much correct, I just hadn't updated yet.

![image](https://user-images.githubusercontent.com/474879/206764990-90977fc5-d1b4-436b-a17b-51d6a0439f26.png)

Loss graph no longer looks cyclic. Good work. 👍
